### PR TITLE
Add test to show that you can't get data for object field defined with `additionalProperties: { ... }`

### PR DESCRIPTION
**Pull request by adam-lynch**
This replaces #4950. While I was off work, someone accidentally deleted our fork which closed the old PR.

### CODE REVIEW

File: test/unit/bug-report.test.ts [hunk 1]
```diff
@@ -83,54 +97,39 @@ describe('bug-report.test.js', () => {
...
          */
-        assert.strictEqual(myDocument.age, 56);
-
-        // you can also wait for events
-        const emitted = [];
-        const sub = collectionInOtherTab.mycollection
-            .findOne().$
-            .subscribe(doc => emitted.push(doc));
-        await AsyncTestUtil.waitUntil(() => emitted.length === 1);
+        assert.strictEqual(myDocument.toJSON().tags, tags);
```
Comment (pubkey) [inline test/unit/bug-report.test.ts:128]: shouldn't this be assert.deepStrictEqual()?

Comment (adam-lynch) [reply test/unit/bug-report.test.ts]: Updated

**Comment by pubkey** [comment]
It is really strange that this only fails on bun but works on Node and Deno

**Comment by pubkey** [comment]
Added the test to the test suite. https://github.com/pubkey/rxdb/pull/5244
I had to use `deepEqual()`, somehow the assert module of bun is broken.

**Pull request closed without merging**